addons/account/models/account_account.py

```diff
@@ -999,37 +1029,80 @@ def write(self, vals):
             for account in self:
                 if self.env['account.move.line'].search_count([('account_id', '=', account.id), ('currency_id', 'not in', (False, vals['currency_id']))]):
                     raise UserError(_('You cannot set a currency on this account as it already has some journal entries having a different foreign currency.'))
-        res = super().write(vals)
-        if {'company_ids', 'code'} & vals.keys():
+
+        if vals.get('code_mapping_ids') is False:
```

Comment:
Can't you have a function like `_sanitize_vals` called in both `create` and `write` to avoid :copyright::spaghetti: ?